src/schema-processor/json/parser.ts

```diff
@@ -223,4 +223,44 @@ export class Parser {
         throw new Error(`field ${field} not specified in serialization info`);
     }
   }
+
+  /**
+   * ExtractMetadata return metadata from JSON schema
+   *
+   * @param {string} schema - JSON schema
+   * @returns SchemaMetadata
+   */
+  public static extractMetadata(schema: string): SchemaMetadata {
+    const parsedSchema = JSON.parse(schema);
+    return parsedSchema.$metadata;
+  }
+
+  public static async getPossibleCredenitalTypesForJsonSchema(
```

Comment:
typo - getPossibleCredent **i** alTypesForJsonSchema

but anyway. logic is not correct.
we should have a function getCredenitalTypeForJsonSchema
and inside you need to filter types based on fields inside credentialSubject in the JSON schema.

e.g. if it has a field 'salary' type 'KYCAgeCredential' must not be returned because it doesn't have a field salary in its context.